submissions/antonkottans/tjsw-post-oop/index.js

```diff
@@ -0,0 +1,166 @@
+const KEYS_VALUE_SEPARATOR = ": ";
+const PROPS_SEPARATOR = "; ";
+
+class Creature {
+  constructor({ name, gender, friends = [], species, saying }) {
+    this.name = name;
+    this.species = species;
+    this.gender = gender;
+    this.friends = friends;
+    this.saying = saying;
+  }
+
+  createArrayOfProps() {
+    return [
+      this.name,
+      this.species,
+      this.gender,
+      this.saying,
+      this.friendsToString(),
+    ];
+  }
+
+  friendsToString() {
+    return this.friends.length
+      ? this.friends.map((friend) => friend.name).join(", ")
+      : "no friends";
+  }
+
+  addFriends(...newFriends) {
+    return (this.friends = [...this.friends, ...newFriends]);
+  }
+}
+
+class Animal extends Creature {
+  constructor({ name, gender, friends, paws, species, saying }) {
+    super({ name, gender, friends, species, saying });
+    this.paws = paws;
+  }
+
+  createArrayOfProps() {
+    return [...super.createArrayOfProps(), this.paws];
+  }
+}
+
+class AnimalWithTail extends Animal {
+  constructor({ name, gender, friends, paws, species, saying, tail }) {
+    super({ name, gender, friends, paws, species, saying, tail });
```

Comment:
What `super` essentially is and what it does?

Reply:
its a function, that calls parent's constructor i don't need to pass in a tail to super?